internal/k8s/controllers/tcp_route_controller_test.go

```diff
@@ -0,0 +1,170 @@
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gw "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/gatewayclient"
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/gatewayclient/mocks"
+	reconcilerMocks "github.com/hashicorp/consul-api-gateway/internal/k8s/reconciler/mocks"
+	"github.com/hashicorp/go-hclog"
+)
+
+var (
+	tcpRouteName = types.NamespacedName{
+		Name:      "tcp-route",
+		Namespace: "default",
+	}
+)
+
+func TestTCPRouteSetup(t *testing.T) {
+	require.Error(t, (&TCPRouteReconciler{}).SetupWithManager(nil))
+}
+
+func TestTCPRoute(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		name          string
+		err           error
+		result        reconcile.Result
+		expectationCB func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager)
+	}{{
+		name: "get-error",
+		err:  errExpected,
+		expectationCB: func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager) {
+			client.EXPECT().GetTCPRoute(gomock.Any(), tcpRouteName).Return(nil, errExpected)
+		},
+	}, {
+		name: "deleted",
+		expectationCB: func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager) {
+			client.EXPECT().GetTCPRoute(gomock.Any(), tcpRouteName).Return(nil, nil)
+			reconciler.EXPECT().DeleteTCPRoute(gomock.Any(), tcpRouteName)
+		},
+	}} {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			client := mocks.NewMockClient(ctrl)
+			reconciler := reconcilerMocks.NewMockReconcileManager(ctrl)
+			if test.expectationCB != nil {
+				test.expectationCB(client, reconciler)
+			}
+
+			controller := &TCPRouteReconciler{
+				Context:        context.Background(),
+				Client:         client,
+				Log:            hclog.NewNullLogger(),
+				ControllerName: mockControllerName,
+				Manager:        reconciler,
+			}
+			result, err := controller.Reconcile(context.Background(), reconcile.Request{
+				NamespacedName: tcpRouteName,
+			})
+			if test.err != nil {
+				require.Error(t, err)
+				require.ErrorIs(t, err, test.err)
```

Comment:
#nitpick I assume `ErrorIs` will check to ensure `err` is not nil, so probs don't need both

Reply:
Looked at the source and yea this looks safe, `ErrorIs` check seems sufficient.
```suggestion
				require.ErrorIs(t, err, test.err)
```